includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -18,51 +18,85 @@ class AMP_Img_Sanitizer extends AMP_Base_Sanitizer {
 	private static $script_src = 'https://cdn.ampproject.org/v0/amp-anim-0.1.js';
 
 	public function sanitize() {
+
 		$nodes = $this->dom->getElementsByTagName( self::$tag );
+		$have_dimensions = array();
+		$need_dimensions = array();
+
 		$num_nodes = $nodes->length;
+
 		if ( 0 === $num_nodes ) {
 			return;
 		}
 
 		for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
 			$node = $nodes->item( $i );
-			$old_attributes = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $node );
 
-			if ( empty( $old_attributes['src'] ) ) {
+			if ( ! $node->hasAttribute( 'src' ) || '' === $node->getAttribute( 'src' ) ) {
 				$node->parentNode->removeChild( $node );
 				continue;
 			}
 
-			$new_attributes = $this->filter_attributes( $old_attributes );
-
-			// Try to extract dimensions for the image, if not set.
-			if ( ! isset( $new_attributes['width'] ) || ! isset( $new_attributes['height'] ) ) {
-				$dimensions = AMP_Image_Dimension_Extractor::extract( $new_attributes['src'] );
-				if ( is_array( $dimensions ) ) {
-					$new_attributes['width'] = $dimensions[0];
-					$new_attributes['height'] = $dimensions[1];
-				}
+			// Determine which images need their dimensions determined/extracted.
+			if ( ! $node->hasAttribute( 'width' ) || ! $node->hasAttribute( 'height' ) ) {
```

Comment:
An img with attributes `width="" height=""` will end up in the else clause. (Although, I guess this is an existing bug and maybe better solved in a separate issue/PR).